Tests/SwiftSyntaxMacroExpansionTest/AttributeRemoverTests.swift

```diff
@@ -10,505 +10,484 @@
 //
 //===----------------------------------------------------------------------===//
 
-//==========================================================================//
-// IMPORTANT: The macros defined in this file are intended to test the      //
-// behavior of MacroSystem. Many of them do not serve as good examples of   //
-// how macros should be written. In particular, they often lack error       //
-// handling because it is not needed in the few test cases in which these   //
-// macros are invoked.                                                      //
-//==========================================================================//
-
-import SwiftDiagnostics
+import SwiftParser
 import SwiftSyntax
-import SwiftSyntaxMacroExpansion
-import SwiftSyntaxMacros
-import SwiftSyntaxMacrosTestSupport
 import XCTest
-
-fileprivate struct NoOpPeerMacro: PeerMacro {
-  static func expansion(
-    of node: AttributeSyntax,
-    providingPeersOf declaration: some DeclSyntaxProtocol,
-    in context: some MacroExpansionContext
-  ) throws -> [DeclSyntax] {
-    return []
+import _SwiftSyntaxTestSupport
+@_spi(AttributeRemover) import SwiftSyntaxMacroExpansion
+
+fileprivate func assertSyntaxRemovingTestAttributes(
+  of originalSource: () throws -> String,
+  reduction expectedReducedSource: () -> String,
```

Comment:
Any reason why you made theses closures instead of normal arguments?

Reply:
Yes, my intent is to make the resulting tests easier to read:

```swift
// string parameters
assertSyntaxRemovingTestAttributes(
  """
  struct Foo {
    @Test @State var x: Int
  }
  """,
  reduction: """
    struct Foo {
      @State var x: Int
    }
    """
)
```

vs.

```swift
// closure parameters
assertSyntaxRemovingTestAttributes {
  """
  struct Foo {
    @Test @State var x: Int
  }
  """
} reduction: {
  """
  struct Foo {
    @State var x: Int
  }
  """
}
```

I think the use of closure syntax improves readability and better supports visual comparison. There are already 30 test cases using the helper, and the benefits would increase over time as more tests are added. (I've already come up with some missing tests to add, some of which are failing.) However, I recognize this is subjective, and this style may well be out of place in this project. With that considered, let me know how you want me to proceed.

Reply:
I disagree here that the trailing closures are easier to read. I personally prefer normal arguments, which would also be more consistent with eg. `assertParse`, which also takes several multi-line string literals.

Reply:
I refactored the `assertSyntaxRemovingTestAttributes` helper to use String arguments.

Reply:
Thank you!